mediacapture-streams/MediaStreamTrack-getSettings.https.html

```diff
@@ -8,72 +8,78 @@
+    assert_equals(settings1.deviceId, settings2.deviceId);
   }, 'A device can be opened twice and have the same device ID');
 
-  promise_test(t => {
+  promise_test(async t => {
     const constraints = {
       video: true,
       audio: false
     };
 
-    return navigator.mediaDevices.getUserMedia(constraints)
-    .then(mediaStream => {
-      const settings1 = mediaStream.getVideoTracks()[0].getSettings();
-      const videoConstraints = {
-        deviceId: settings1.deviceId,
-        width: {
-          exact: settings1.width / 2
-        }
-      };
-
-      return navigator.mediaDevices.getUserMedia({
-        video: videoConstraints,
-        audio: false
-      }).then(mediaStream => {
-        const settings2 = mediaStream.getVideoTracks()[0].getSettings();
-        assert_equals(settings1.deviceId, settings2.deviceId);
-        assert_equals(settings1.width / 2, settings2.width);
-      });
-    });
-  }, 'A device can be opened twice with different resolutions');
+    const stream1 = await navigator.mediaDevices.getUserMedia(constraints);
+    const tracks = stream1.getTracks();
+    t.add_cleanup(() => tracks.forEach(track => track.stop()));
 
-  promise_test(t => {
-    return navigator.mediaDevices.enumerateDevices().then(async devices => {
-      for (var device of devices) {
-        if (device.kind == "audiooutput")
-          continue;
-        var device_id_constraint = {deviceId: {exact: device.deviceId}};
-        var constraints = device.kind == "audioinput"
-          ? {audio: device_id_constraint}
-          : {video: device_id_constraint};
-
-        var stream = await navigator.mediaDevices.getUserMedia(constraints);
-        assert_equals(stream.getTracks()[0].getSettings().groupId,
-                      device.groupId);
-        assert_greater_than(device.groupId.length, 0);
+    const settings1 = stream1.getVideoTracks()[0].getSettings();
+    const videoConstraints = {
+      deviceId: settings1.deviceId,
+      width: {
+        exact: settings1.width / 2
       }
+    };
+
+    const stream2 = await navigator.mediaDevices.getUserMedia({
+      video: videoConstraints,
+      audio: false
     });
+    tracks.push(...stream2.getTracks());
+
+    const settings2 = stream2.getVideoTracks()[0].getSettings();
+    assert_equals(settings1.deviceId, settings2.deviceId);
+    assert_equals(settings1.width / 2, settings2.width);
+  }, 'A device can be opened twice with different resolutions');
+
+  promise_test(async t => {
+    const devices = await navigator.mediaDevices.enumerateDevices();
+    const tracks = [];
+    t.add_cleanup(() => tracks.forEach(track => track.stop()));
+    for (const device of devices) {
+      if (device.kind == "audiooutput")
+        continue;
+      const device_id_constraint = {deviceId: {exact: device.deviceId}};
+      const constraints = device.kind == "audioinput"
+        ? {audio: device_id_constraint}
+        : {video: device_id_constraint};
```

Comment:
This is going to do the wrong thing for `audiooutput`.

Reply:
The `audiooutput` has already been ruled out in L70~L71.